homeassistant/components/vera.py

```diff
@@ -57,35 +57,42 @@ def setup(hass, base_config):
     global VERA_CONTROLLER
     import pyvera as veraApi
 
-    config = base_config.get(DOMAIN)
-    base_url = config.get(CONF_CONTROLLER)
-    VERA_CONTROLLER, _ = veraApi.init_controller(base_url)
-
     def stop_subscription(event):
         """Shutdown Vera subscriptions and subscription thread on exit."""
         _LOGGER.info("Shutting down subscriptions.")
         VERA_CONTROLLER.stop()
 
+    config = base_config.get(DOMAIN)
+
+    # Get Vera specific configuration.
+    base_url = config.get(CONF_CONTROLLER)
+    light_ids = config.get(CONF_LIGHTS)
+    exclude_ids = config.get(CONF_EXCLUDE)
+
+    # Initialize the Vera controller.
+    VERA_CONTROLLER, _ = veraApi.init_controller(base_url)
     hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, stop_subscription)
 
     try:
-        all_devices = VERA_CONTROLLER.get_devices()
+        devices = VERA_CONTROLLER.get_devices()
```

Comment:
Could you switch this back to `all_devices`? I think it makes the code a little clearer to avoid mutating `devices`